src/utils/time.ts

```diff
@@ -131,76 +170,185 @@ export function shortTime(el: timeElement): timeElement {
   };
 }
 
-export function timeToString(el: timeElement): string {
-  let output = '';
+export function timestampToTime(timestamp: number): { time: durationFormat; isFuture: boolean } {
+  const map: durationFormat = {};
+  const isFuture = timestamp > Date.now();
+  let timestampAbs = Math.abs(timestamp - Date.now());
 
-  if (el.y === 1) {
-    output += ` ${el.y} ${api.storage.lang('bookmarksItem_Year')}`;
-  } else if (el.y > 1) {
-    output += ` ${el.y} ${api.storage.lang('bookmarksItem_Years')}`;
+  for (const key in dateUnitToMs) {
+    const value = Math.floor(timestampAbs / dateUnitToMs[key]);
+    map[`${key}s`] = value;
   }
+  const mapKeys = Object.keys(map);
+  for (let i = 1; i < mapKeys.length; i++) {
+    const keyPrev = mapKeys[i - 1];
+    const valuePrev = map[keyPrev];
+    const keyCurr = mapKeys[i];
 
-  if (el.d === 1) {
-    output += ` ${el.d} ${api.storage.lang('bookmarksItem_Day')}`;
-  } else if (el.d > 1) {
-    output += ` ${el.d} ${api.storage.lang('bookmarksItem_Days')}`;
+    timestampAbs = timestampAbs - dateUnitToMs[keyPrev.slice(0, -1)] * valuePrev;
+    const valueCurr = Math.floor(timestampAbs / dateUnitToMs[keyCurr.slice(0, -1)]);
+    map[keyCurr] = valueCurr;
   }
+  return { time: map, isFuture };
+}
 
-  if (el.h === 1) {
-    output += ` ${el.h} ${api.storage.lang('bookmarksItem_Hour')}`;
-  } else if (el.h > 1) {
-    output += ` ${el.h} ${api.storage.lang('bookmarksItem_Hours')}`;
-  }
+// TODO: Delete @ts-expect-error comments after TS adds support for Intl.DurationFormat
+export function getDurationInLocale(
+  duration: durationFormat,
+  locale: Intl.LocalesArgument = api.storage.lang('locale'),
+  style: durationFormatStyle = 'narrow',
+): string {
+  const minsSecondsOnly =
+    (Object.keys(duration).length === 2 && duration.minutes && duration.seconds) ||
+    (Object.keys(duration).length === 1 && (duration.minutes || duration.seconds));
 
-  if (el.m === 1) {
-    output += ` ${el.m} ${api.storage.lang('bookmarksItem_min')}`;
-  } else if (el.m > 1) {
-    output += ` ${el.m} ${api.storage.lang('bookmarksItem_mins')}`;
-  }
+  const inputDuration = minsSecondsOnly
+    ? minsSecsToHoursMins(duration.minutes, duration.seconds)
+    : duration;
```

Comment:
Should not be handled here. Should be a task for getProgressDateTimeInLocale add different templates for shortTime time if necesseray

Reply:
But the progress are only accepts timestamp so no problem with mins. If I'm moving it, how about remove it from here and just call this `minsSecsToHoursMins` function inside meta files to convert to proper hh:mm?

Reply:
It adds unneccessary logic to the Intl.DurationFormat wrapper. I see the handling of the time as a 3 steps process.

1. transformer -> Transforms timestamps to durationFormat -> this would just complely remove the need for minsSecsToHoursMins as you can just pass it as a seconds timestamp.
2. formater -> example.: shortTime -> Which takes a durationFormat and takes the hour etc we want to display.
3. Renderer -> Intl.DurationFormat wrapper

Combining 1. and 2. is fine by my book. But 3. should be seperate, as it results in unnecessary amount of different variantations.

Reply:
Alternative we could think about going for OOP. Then it would make it simpler to handle the different inputs and output formats. But no idea
`obj.setTimestamp(timestamp)->format('progress')`
`obj.setDate(timestamp)->getDateInLocale()`

Reply:
The thing is - `getDurationInLocale` is different from other functions since it just takes `durationFormat` and parses it to string. And I cant just pass it ms units because in the end it will be like this "Duration: 4500000 ms". I can add "transformer" function inside - `timestampToTime` but it converts timestamp relatively to current time. I guess I can do something like this -> Pass (ms duration of anime + `Date.now`) to `timestampToTime`, it calculates how much "time will pass" and returns `durationFormat`, but doesn't it feel like we using function out of purpose?

And yes I agree with your 3 points. As for turning it into OOP idk really, I'll sit for a bit and try something out.